### DISCUSSION TITLE
Nueva funcionalidad dentro de Google Oauth y tests

### DISCUSSION THREAD

**Discussion by MarioArocaPaez** [perealberto/decide-single-zambrano#29] [Q&A] [answered]
Hola gente, unas duditas 🎅

Para el tema de registrarte con Google he visto que una cosa guay que se le puede añadir es mandar un correo al gmail de quien se registra cuando lo haga. Una tontería rollo bienvenido a decide y tal, pero creo que está chulo y con alguna API se puede hacer medianamente fácil ¿Qué os parecería si dedico una tarde a implementarlo? 🙏

Luego como fuimos medio apresurados para el M2 me gustaría añadir tests tanto para la función de registrarse por el formulario como la de Google, entonces mi pregunta es. Queriendo hacer este enhancement a lo de Goolge y también queriendo añadirle mas tests ¿Como debería proceder? ¿Creo una nueva rama? ¿Uso la rama de la propia funcionalidad pero la actualizo antes con develop? ¿Como tratamos esto?

**Comment by laurolmer** [comment, marked as answer]
Estimado compañero, me parecen buenas alternativas y más de cara al próximo M3 al que el equipo se deberá enfrentar. En mi humilde opinión, creo que deberías generar una nueva rama con la etiqueta "test" y ahí involucrar esas pruebas. Asimismo, creo que deberías crear una issue respectiva a ella y ya proceder de manera normal y corriente con las pull-requests, etc.

Espero que te haya servido fiera

**Reply by MarioArocaPaez** [reply]
![imagen](https://github.com/3ralon/decide-single-zambrano/assets/73229438/150b84be-9054-4afb-b51a-5e05cbb2d040)

**Comment by perealberto** [comment]
Buenas a todos 😎 

En cuanto a la nueva funcionalidad. Me parece muy chula, por mí sin problemas. Creo que podrías reutilizar la rama que ya tenías porque al final es una extensión de la funcionalidad.

Luego, en cuanto a los tests, se me ocurren dos cosas:

- Cuando se suba un cambio a develop, seleccionamos a una persona que no haya hecho esa funcionalidad para que la pruebe. Se crea una rama para hacer los tests con el formato `Test/Nombre_funcionalidad` y ya luego se haría *merge* a main cuando se compruebe que todo está correcto.

- Al igual que en la anterior, seleccionamos una persona para que realice los tests. Pero, en lugar de crear una nueva rama, los tests se realizan en la rama de la funcionalidad.

Creo que vendría bastante bien que pongamos a hacer test de una funcionalidad a las personas que no hayan tocado esa parte del proyecto, así todos tenemos que estar enterados de todos los cambios.

_MSProject is temporary, **spring is eternal**_ 🍃

**Comment by MarioArocaPaez** [comment]
No se ha podido realizar por falta de tiempo :(
